hydra-nl/process_hydranl.py

```diff
@@ -0,0 +1,127 @@
+                            hydra_results['HYD_location_name'].append(invoer['algemeen']['LOCATIE'])
+                            hydra_results['HYD_location_X'].append(invoer['algemeen']['XCOORDINAAT'])
+                            hydra_results['HYD_location_Y'].append(invoer['algemeen']['YCOORDINAAT'])
+                            hydra_results['Profiel'].append(valueprofiel)
+                            
+                            # get ondergrens doorsnede-eis for a specific vak from excel
+                            for i in range(len(okader_norm_csv['VAK'])):
+                                if okader_norm_csv['VAK'].loc[i] == int(value):
+                                    req_terugkeertijd = okader_norm_csv['T_eis_hbn'].loc[i]
+
+
+                        # based on uitvoer.html (bij meest waarschijnlijke combinatie van illustratiepunt) the other columns of our dictionairy can be filled
+                        elif file.endswith('uitvoer.html'):
+                            location_uitvoer = os.path.join(root, file)
+                            tekst = lees_uitvoerhtml(location_uitvoer)
+                            cips, ip = lees_illustratiepunten(tekst)
+                            ofl = lees_ofl(tekst)
+
+                            # check if this results in 2 illustratiepunten, only one is needed in dictionairy
+                            a = (ip['bijdrage ov. freq (%)'].loc[[req_terugkeertijd][0]])
+                            if a.size > 1:
+                                rel_ip = ip.loc[[req_terugkeertijd][0]]
+                                rel_ip = rel_ip.iloc[np.argmax(rel_ip['h,teen m+NAP'])]
+                            else:
+                                rel_ip = ip.loc[[req_terugkeertijd][0]]
+                        
+                            hydra_results['Terugkeertijd'].append(req_terugkeertijd)
+
+                            # append the results to the dictionairy of the most probable combination in the illustratiepunt at the doorsnede eis ondergrens at the Okader VakId
+                            hydra_results['HBN [m+NAP]'].append(ofl['hoogte'].loc[[req_terugkeertijd][0]])
+                            hydra_results['Zeewaterstand [m+NAP]'].append(rel_ip['zeews. m+NAP'])
+                            hydra_results['windsnelheid [m/s]'].append(rel_ip['windsn. m/s'])
+                            hydra_results['h, teen [m+NAP]'].append(rel_ip['h,teen m+NAP'])
+                            hydra_results['Hm0, teen [m]'].append(rel_ip['Hm0,teen m'])
+                            hydra_results['Tm-1.0, ts'].append(rel_ip['Tm-1,0,t s'])
+                            hydra_results['golfrichting [graden]'].append(rel_ip['golfr graden'])
+                            hydra_results['windrichting [graden N]'].append(rel_ip['r'])
+                            hydra_results['bijdrage ov. freq [%]'].append(rel_ip['bijdrage ov. freq (%)'])
+
+# make dataframe for saving as a csv          
+df = pd.DataFrame(hydra_results)
```

Comment:
The problem with this method is that if a folder does not contain an 'invoer.hyd' and an 'uitvoer.hyd' file, then the dataframe cannot be created as the lists in the dictionary are not the same length. 

If you're sure that these files are present, then just open them directly instead of looping through all the files in the folder. If the file does not exist, raise an error.